src/AzureAuth/Commands/CommandAad.cs

```diff
@@ -405,5 +418,34 @@ private int GetToken(IPublicClientAuth publicClientAuth)
 
             return 0;
         }
+
+        /// <summary>
+        /// Sets the <see cref="AuthMode"/> from the environment variable and sets a default if not set.
+        /// </summary>
+        /// <returns>True if authmode is set.</returns>
+        public bool TrySetAuthModeFromEnvOrDefault()
+        {
+            var authModesFromEnv = this.env.Get(EnvVars.AuthMode);
+            if (string.IsNullOrEmpty(authModesFromEnv))
+            {
+                this.AuthModes = new[] { AuthMode.Default };
+            }
+
+            var result = new List<AuthMode>();
+            foreach(var val in authModesFromEnv.Split(','))

```

Comment:
Are commas or semi-colons more typical for env variable separation? I think the Path variable separates multiple values with semi-colons.

Reply:
oh yes, I thought about this. But I went with the same format that option --mode allows which is comma-separated, so that the input format will be consistent.

Reply:
I think that's reasonable :) I don't know much about env variable conventions, but from searching around, it sounds like environment variables that map to directories/paths should use semi-colon delimiters. However, for other environment variables, it appears there isn't really a convention.

So I think the way you have it is great, and your justification makes perfect sense.